ci-operator/step-registry/ipi/deprovision/aws/sharednetwork/ipi-deprovision-aws-sharednetwork-commands.sh

```diff
@@ -7,20 +7,34 @@ trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wa
 
 export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
 
-if test ! -f "${SHARED_DIR}/sharednetworkstackname"
-then
-  echo "No sharednetworkstackname, so unknown stack name, so unable to tear down."
-  exit 0
-fi
-
 REGION="${LEASED_RESOURCE}"
-STACK_NAME="$(cat "${SHARED_DIR}/sharednetworkstackname")"
 
-# cleaning up after ourselves
-aws --region "${REGION}" cloudformation delete-stack --stack-name "${STACK_NAME}" &
-wait "$!"
+function delete_stack() {
+  local stack_file_name
+  stack_file_name=$1; shift
+  exit_on_missing=${1:-"no"}
+
+  if test ! -f "${SHARED_DIR}/${stack_file_name}"
+  then
+    echo "No ${stack_file_name}, so unknown stack name, so unable to tear down."
+    test "${exit_on_missing}" == "yes" && exit 0
```

Comment:
what is the point of having `exit_on_missing` == no? if the stack_name can't be found, won't the following code always fail and therefore we should exit?

Reply:
I am trying to keep the same behavior [as the original code](https://github.com/openshift/release/pull/39902/files#diff-8816d4bda929f689adbde65df4c75c5cbed38fc6c8db3b514f10ef97e1063009L10-L14). I am not sure if it was added. Maybe when resource pruned removed the Stack and prevent failures in CI?! (just guessing)

Reply:
@patrickdillon Sorry I misunderstood when I read your comment the last time.

Now I got what you are saying, and I changed [here](https://github.com/openshift/release/compare/9d590f05b08737652c911ebf192e6d8351a5cd58..8879c2534fac876691770450d42d703d3094c9a3#diff-8816d4bda929f689adbde65df4c75c5cbed38fc6c8db3b514f10ef97e1063009R21). When the Local Zone stack does not exist (file not found: `exit_on_missing == no`), it will continue the flow and delete the VPC.

Reply:
Maybe that extra arg, and exiting, is unnecessary right now. So we can always return when the stack file name is not present, keeping the same behavior of the original code when not using a function.